libgalois/include/katana/PagePool.h

```diff
@@ -105,6 +119,8 @@ class PageAllocState {
 
   void pageFree(void* ptr) {
     KATANA_LOG_DEBUG_ASSERT(ptr);
+    auto tid = katana::ThreadPool::getTID();
+    freeCounts[tid] += 1;
```

Comment:
`getTID` is a little more expensive than a global variable access so it is good to avoid it if we can.

In this case, `tid` is the same as `i` below, so perhaps change Line 126 to `int tid = ownerMap[ptr]` and just moving `freeCounts[tid] += 1` down.